lib/install/deps.js

```diff
@@ -600,6 +600,7 @@ var earliestInstallable = exports.earliestInstallable = function (requiredBy, tr
   var binaryMatches = typeof pkg.bin === 'object' && tree.children.some(function (child) {
     if (child.removed) return false
     if (typeof child.package.bin !== 'object') return false
+    if (!pkg.bin || !child.package.bin) return false
```

Comment:
In my latest commit I changed `typeof pkg.bin === 'object' && ...` into `pkg.bin && ...`. 👍 

@iarna @watilde Can you review it and help me to improve the tests? 💭